synapse/http/matrixfederationclient.py

```diff
@@ -339,9 +341,26 @@ def _send_request(
         else:
             query_bytes = b""
 
-        headers_dict = {b"User-Agent": [self.version_string_bytes]}
+        # Retreive current span
+        scope = opentracing.tracer.start_active_span(
+            "outgoing-federation-request",
+            tags={
+                "span.kind": "client",  # With respect to this request's role in an rpc
+                "peer.address": request.destination,
+                "http.method": request.method,
+                "http.url": request.path,
```

Comment:
For these constants you probably want to use `tags.SPAN_KIND`, `tags.SPAN_KIND_RPC_CLIENT` etc from `opentracing.ext`. This just prevents accidental typos (which I do a lot :/)